File: simol_framework/simol_php_framework/org/simol/framework/php/Simu_real_world_obj.php
<?php
namespace org\simol\framework\php;

use \Attribute;

enum Role /* since PHP 8.1 */
{
    case DESIGN_ROLE;
    case HUMAN_EXPERT;
    case EXPERT_COMPONENT;
    case GUI_ROLE;
    case MACHINE_TOOL;
    case UNDEFINED;
};

/**
 * Simol framework for PHP 8. v1.5(20240319).
 * 
 * Renamed Simu_comp_worker to Simu_comp_as_worker. 
 * 
 * About Simol PHP framework:
 * 
 * Nature and Constracts not present due to PHP attributes limitations, and
 * With_compat is not on local variables.
 * TurnMonitor, For_every_thread, One_at_a_time, Dependent_activity because thread support is not supplied in Simol PHP Framework.
 */

#[Attribute(Attribute::TARGET_CLASS)]
class Simu_real_world_obj
{
    
    private $role;
    
    public function __construct(Role $theRole = Role::UNDEFINED)
    {
        $this->role = $theRole;
    }
}

File: simol_framework/SimolCsharpFramework/SimolCsharpFramework/Simu_real_world_obj.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/***
 * Simol C# Framework, v1.9, 20240319.
 * 
 * Renamed Simu_comp_worker to Simu_comp_as_worker.
 * 
 * 
 * Limitations of C# framework for simol:
 * 
 * No 'nature' or 'contracts' attributes, nor possibility to have 'With_compat' on a local variable. All this is due
 * to C# attributes limitations. And no Forced_inh and forced_int_inh on types of the list of inherited class and interfaces of a class/interface definition.
 */
namespace Org.Simol.Framework.Csharp
{
    public enum Role
    {
        DESIGN_ROLE,
        HUMAN_EXPERT,
        EXPERT_COMPONENT,
        GUI_ROLE,
        MACHINE_TOOL,
        UNDEFINED
    };

    [AttributeUsage(AttributeTargets.Class)]
    public class Simu_real_world_objAttribute: Attribute
    {
       
        private Role role;

        public Simu_real_world_objAttribute(Role theRole = Role.UNDEFINED)
        {
            this.role = theRole;
        }
    }
}
